General Settings/Language: sort items based on descriptions

josephsl opened the pull request:
### Link to issue number:
Fixes #7284

### Summary of the issue:
Items in general settings/language are sorted by language codes.

### Description of how this pull request fixes the issue:
Users may not realize the languages list in General settings/Language combo box are sorted by ISO 639-1 codes. Thus improve user experience by sorting items by descriptions instead, and when possible, just display human-readable language names (not codes).

### Testing performed:
Tested on Windows 10 Version 1803 (17133) with languages set to English and Korean.

### Known issues with pull request:
None

### Change log entry:
Changes: the list of available languages in General Settings dialog is now sorted based on language names instead of ISO 639 codes. (#7284)

josephsl committed:
Language handler: allow presentational view of available languages to be obtained. Re #7284.

When going through General settings/Language combo box, users might be puzzled as to why items in there are listed in random order. This is because entries are sorte according to ISO 639-1 code. To improve user experience, sort this alphabetically according to description.
This is done through a new keyword argument in languageHandler.getAvailableLanguages function. Wuth presentational argument set, language tuples will be sorted based on description text.

josephsl committed:
Settings dialog/General/Language: enable presentational view of languages. Re #7284.

josephsl committed:
GUI/settings dialog: update copyright years

josephsl committed:
Docstring: language handler module description

josephsl committed:
Language handler: only add language descriptions if this is for presentation purposes. Re #7284.

Reduece confusion further by omitting language codes from language description texts.

josephsl commented:
Hi, yes, user default will be the last option. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Thursday, April 5, 2018 12:21 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] General Settings/Language: sort items based on descriptions (#8143)

 

 <https://github.com/josephsl> @josephsl commented on 4 apr. 2018 17:50 CEST <https://github.com/nvaccess/nvda/pull/8143#issue-179434878> :


Link to issue number:


and when possible, just display human-readable language names (not codes).

I tend to disagree here. When I somehow end up at a Korean system running NVDA, all the language names will be in Korean. As I don't know Korean, I will need the ISO code to find the appropriate language.

I haven't tested the code yet, but I assume that the "user default" option is still the last one in the list?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8143#issuecomment-378843409> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkEqIfh2wXF95YIETCpmsb6XzKqTSks5tlcX0gaJpZM4THDI-> .

michaelDCurran commented:
I agree with @leonardder here: please keep the language code in the description.

josephsl committed:
Merge branch 'master' into i7284sortedLangsList

josephsl committed:
Language handler: clarify module docstring

josephsl committed:
Language handler: readability changes for variable namess.

the nanes 'd', 'i', and 'l' are so generic. Thus rename them to 'displayNames', 'entry', and 'locales'.

josephsl committed:
Language handler: clarify comment

josephsl committed:
Language handler: add ISO 639 codes in order to reduce confusion for users. Re #7284.

Reviewed by Mick Curran (NV Access) and others: suppose a user changes to another language by accident, a language he or she cannot understand, especially changes the Windows display language by accident. So how can they return to a language they know? This cannot be done with absence of ISO 639 language codes.

leonardder commented on source/languageHandler.py:
```diff
@@ -94,29 +98,34 @@ def getLanguageDescription(language):
 	return desc
 
-def getAvailableLanguages():
+def getAvailableLanguages(presentational=False):
```
I think alphabetize or sort might be a clearer keyword argument instead of presentational. Furthermore, someone might be interested in getting a sorted list of languages for non presentational purposes.

josephsl commented:
Hi, it is already sorted by language code, so the presentational keyword name makes a bit more sense (as it’ll be used in the UI).

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Monday, April 30, 2018 7:02 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] General Settings/Language: sort items based on descriptions (#8143)

 

@leonardder commented on this pull request.

  _____  

In source/languageHandler.py <https://github.com/nvaccess/nvda/pull/8143#discussion_r184994817> :

> @@ -94,29 +98,34 @@ def getLanguageDescription(language):
               }.get(language,None)
        return desc
 
-def getAvailableLanguages():
+def getAvailableLanguages(presentational=False):

I think alphabetize or sort might be a clearer keyword argument instead of presentational. Furthermore, someone might be interested in getting a sorted list of languages for non presentational purposes.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8143#pullrequestreview-116297027> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkG1goOe0hhQyAzVFEvJqJU4WlA4lks5ttxlFgaJpZM4THDI-> .

josephsl commented:
Hi, “going global” is a descriptive term for “internationalization”. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Monday, April 30, 2018 7:00 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] General Settings/Language: sort items based on descriptions (#8143)

 

@leonardder commented on this pull request.

  _____  

In source/languageHandler.py <https://github.com/nvaccess/nvda/pull/8143#discussion_r184994419> :

>  #This file is covered by the GNU General Public License.
 #See the file COPYING for more details.
 
+"""Language and localization support.
+This module assists in NVDA going global through language services such as converting Windows locale ID's to friendly names and presenting available languages.

What do you mean with going global here?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8143#pullrequestreview-116296511> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkHPOMzZXqFh6eYBqexHZhfRnfQS-ks5ttxjpgaJpZM4THDI-> .
